tests/functional/test_warning.py

```diff
@@ -0,0 +1,36 @@
+import pytest
+
+PY26_WARNING = "Python 2.6 is no longer supported"
+
+
+@pytest.mark.skipif("sys.version_info >= (2,7)")
+def test_python26_options(script):
+    result = script.run(
+        'python', '-m', 'pip.__main__', 'list', expect_stderr=True,
+    )
+    assert PY26_WARNING in result.stderr
+    result = script.run('python', '-W', 'ignore', '-m', 'pip.__main__', 'list')
+    assert result.stderr == ''
+
+
+@pytest.mark.skipif("sys.version_info < (2,7)")
+def test_environ(script, tmpdir):
+    """$PYTHONWARNINGS was added in python2.7"""
+    demo = tmpdir.join('warnings_demo.py')
```

Comment:
Yup, I had something like `pip install some_local_package --no-binary=some_local_package` in mind but indeed, this will only work until `--no-binary` is effectively dropped so your solution seems indeed more future resilient.